virtio-devices/src/balloon.rs

```diff
@@ -94,6 +95,45 @@ pub struct VirtioBalloonConfig {
     actual: u32,
 }
 
+#[derive(Clone, Debug)]
+struct PartiallyBalloonedPage {
+    addr: u64,
+    bitmap: Vec<u64>,
+}
+
+impl PartiallyBalloonedPage {
+    fn new() -> Self {
+        let size = get_page_size();
+        let len = ((size >> VIRTIO_BALLOON_PFN_SHIFT) + 63) / 64;
+        // Initial each padding bit as 1 in bitmap.
+        let mut bitmap = vec![0_u64; len as usize];
+        let pad_num = len * 64 - (size >> VIRTIO_BALLOON_PFN_SHIFT);
+        bitmap[(len - 1) as usize] = !((1 << (64 - pad_num)) - 1);
+        Self { addr: 0, bitmap }
+    }
+
+    fn pfn_match(&self, addr: u64) -> bool {
+        let page_size: u64 = get_page_size();
```

Comment:
I think the code would be simplified if we add the `host_page_size` to the `PartiallyBalloonedPage` struct. WDYT?

Reply:
OK, it's better.